### ISSUE TITLE
[Bug]: Missing Endpoints error on testing config when only hostname and port are bound

### ISSUE THREAD

**Issue by btsgh** [appsmithorg/appsmith#20153]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

When I bind only the host name and port, and after filling all the other details in the DS configuration page, I click on Test Configuration, I get an error - Missing Endpoints - and test does not go through

### Steps To Reproduce

1. Create a new ds, save it, and go to manage environments
2. Now bind only the Host name and Port for Production
3. On the ds page, add the other details like db name, user and password for the above host port combination
4. Click on Test Configuration
5. An error pops up "Missing Endpoints"
No error should pop up - the test should go through fine.

https://ddpa7mfuvv.vmaker.com/record/hYxLvZgD90NzxfNW

### Version

EC2 instance

**Comment by ayushpahwa** [comment]
@sondermanish can you take a look into this?

**Comment by sondermanish** [comment]
@ayushpahwa this happens due to id being not sent along with datasourceConfiguration. 
This has same root cause as #20262

**Comment by ayushpahwa** [comment]
Will be fixed with #20262

**Comment by ayushpahwa** [comment]
Hey @btsgh @AnandShahs123 this seems like due to an issue that exists in our production also. I have added more details for this on our slack thread here: https://theappsmith.slack.com/archives/C040LHZN03V/p1678177322347259.

**Comment by btsgh** [comment]
Checked out the slack thread and the issue you mentioned there @ayushpahwa  - in that issue though you are entering an incorrect value in the db field and it gives missing password error. However, here in this bug that we have logged, it is about missing endpoint when bound values are present for host name and port. These two seem to be different cases, since when I try the same flow on Prod of creating a new DS, and testing, it goes through ok.
